daemon/src/identify/dialer.rs

```diff
@@ -0,0 +1,172 @@
+use crate::identify::protocol;
+use crate::identify::PeerInfo;
+use crate::identify::PROTOCOL;
+use async_trait::async_trait;
+use std::collections::HashMap;
+use tokio::sync::watch;
+use tokio_extras::spawn_fallible;
+use xtra::Address;
+use xtra::Context;
+use xtra_libp2p::endpoint;
+use xtra_libp2p::libp2p::PeerId;
+use xtra_libp2p::Endpoint;
+use xtra_libp2p::OpenSubstream;
+use xtra_productivity::xtra_productivity;
+
+pub struct Actor {
+    endpoint: Address<Endpoint>,
+    peer_infos: HashMap<PeerId, PeerInfo>,
+    peer_info_channel: Option<watch::Sender<Option<PeerInfo>>>,
+}
+
+impl Actor {
+    pub fn new(endpoint: Address<Endpoint>) -> Self {
+        NUM_LIBP2P_CONNECTIONS_GAUGE.reset();
+
+        Self {
+            endpoint,
+            peer_infos: HashMap::default(),
+            peer_info_channel: None,
+        }
+    }
+
+    pub fn new_with_subscriber(
+        endpoint: Address<Endpoint>,
+    ) -> (Self, watch::Receiver<Option<PeerInfo>>) {
+        NUM_LIBP2P_CONNECTIONS_GAUGE.reset();
+
+        let (sender, receiver) = watch::channel::<Option<PeerInfo>>(None);
+
+        (
+            Self {
+                endpoint,
+                peer_infos: HashMap::default(),
+                peer_info_channel: Some(sender),
+            },
+            receiver,
+        )
+    }
+}
+
+#[async_trait]
+impl xtra::Actor for Actor {
+    type Stop = ();
+
+    async fn stopped(self) -> Self::Stop {}
+}
+
+pub(crate) struct GetPeerInfo(pub PeerId);
+
+pub(crate) struct IdentifyMsgReceived {
+    peer_id: PeerId,
+    identify_msg: protocol::IdentifyMsg,
+}
+
+#[xtra_productivity]
+impl Actor {
+    async fn handle(&mut self, msg: GetPeerInfo) -> Option<PeerInfo> {
```

Comment:
We should be able to remove this - if I forgot to remove it on #2383 then that was by accident.
After the channel was introduced this is not used.